network/controller/bootstrap/bootstrap.go

```diff
@@ -538,7 +552,16 @@ func (bc *bootstrapper) startBootstrap(ctx context.Context, address string) (*ne
 	if err != nil {
 		return nil, errors.Wrapf(err, "Failed to ping address %s", address)
 	}
-	request := bc.Transport.NewRequestBuilder().Type(types.Bootstrap).Data(&NodeBootstrapRequest{}).Build()
+	claim, err := bc.NodeKeeper.GetOriginJoinClaim()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get a join claim")
+	}
+	bootstrapReq := &NodeBootstrapRequest{
+		// Certificate:   bc.Certificate,
+		JoinClaim:     *claim,
+		LastNodePulse: bc.GetLastPulse(),
```

Comment:
Get/Set LastPulse are workaround methods to prevent races in discovery bootstrap and should be removed in future. Don't use them. Get last pulse from PulseStorage.

Reply:
done